diff --git a/openlayers.module b/openlayers.module
index e51e145..ab34aff 100644
--- a/openlayers.module
+++ b/openlayers.module
@@ -1525,6 +1525,13 @@ function openlayers_libraries_info() {
       'js' => array('OpenLayers.js'),
     ),
     'variants' => array(
+      'original' => array(
+        'files' => array(
+          'js' => array(
+            'OpenLayers.js'
+          ),
+        ),
+      ),
       'original debug' => array(
         'files' => array(
           'js' => array(